mmcv/runner/hooks/logger/wandb.py

```diff
@@ -46,7 +47,20 @@ def before_run(self, runner):
         
         if self.config_path is not None:
             shutil.copy2(self.config_path, self.wandb.run.dir)
-
+            if os.path.isdir(self.config_path):
+                for path, _, _ in os.walk(self.wandb.run.dir):
+                    self.wandb.save(
+                        glob_str=path + "/*",
+                        base_path=self.wandb.run.dir,
+                        policy="now"
+                    )
+            else:
+                self.wandb.save(
+                    glob_str=self.wandb.run.dir + "/*",
+                    base_path=self.wandb.run.dir,
+                    policy="now"
+                )
+    
```

Comment:
This part is added to use wandb.save(). we can upload our config with just copy of config dir, but in this case the uploading occurs at the end of the process. Now, using this lines, we can upload config at the start of training.